backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -0,0 +1,394 @@
+        record_key = (checkout.origin, platform, checkout.id)
+
+        if record_key not in hardware_status_data:
+            hardware_status_data[record_key] = _init_status_record(
+                checkout, platform, test.compatible
+            )
+
+        status_record = hardware_status_data[record_key]
+
+        _process_test_status(
+            test, h_key, status_record, existing_processed, new_processed_entries
+        )
+
+        _process_build_status(
+            build,
+            h_key,
+            status_record,
+            existing_processed,
+            processed_builds_in_batch,
+            new_processed_entries,
+        )
+
+    return hardware_status_data, new_processed_entries
+
+
+class Command(BaseCommand):
+    help = "Process pending tests and builds for hardware status aggregation"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--batch-size",
+            type=int,
+            default=1000,
+            help="Number of pending items to process in one batch",
+        )
+        parser.add_argument(
+            "--loop",
+            action="store_true",
+            help="Run continuously in a loop",
+        )
+        parser.add_argument(
+            "--interval",
+            type=int,
+            default=60,
+            help="Sleep interval in seconds when running in loop mode",
+        )
+
+    def handle(self, *args, **options):
+        batch_size = options["batch_size"]
+        loop = options["loop"]
+        interval = options["interval"]
+
+        if loop:
+            self.stdout.write(
+                f"Starting pending aggregation processor (interval={interval}s)..."
+            )
+            try:
+                while True:
+                    processed_count = self.process_pending_batch(batch_size)
+                    if processed_count == 0:
+                        self.stdout.write(f"Sleeping for {interval} seconds")
+                        time.sleep(interval)
+            except KeyboardInterrupt:
+                self.stdout.write("Stopping pending aggregation processor...")
+        else:
+            self.process_pending_batch(batch_size)
+
+    def process_pending_batch(self, batch_size: int) -> int:
+        last_processed_test_id = None
+
+        while True:
+            self.stdout.write(
+                f"Starting batch processing "
+                f"(last_processed_test_id={last_processed_test_id}, "
+                f"batch_size={batch_size})..."
+            )
+
+            qs = PendingTest.objects.order_by("test_id")
+            if last_processed_test_id:
+                qs = qs.filter(test_id__gt=last_processed_test_id)
+
+            pending_tests_batch = list(qs[:batch_size])
+
+            if not pending_tests_batch:
+                self.stdout.write("No pending tests found, exiting batch")
+                return 0
+
+            last_processed_test_id = pending_tests_batch[-1].test_id
```

Comment:
it's better to set this after the processing loop ends, or during the loop... so you don't need to -1 here